drivers/regulator/axp803.c

```diff
@@ -0,0 +1,532 @@
+		.value_register  = 0x1A,
+		.enable_register = OUTPUT_POWER_CONTROL_2,
+		.bit = 1,
+	},
+	[AXP803_REGL_ELDO3] = {
+		.info = {
+			.name      = "ELDO3",
+			.min_value = 700,
+			.max_value = 1900,
+			.ranges    = {
+				{
+					.start_raw   = 0x00,
+					.start_value = 700,
+					.step        = 50,
+				},
+			},
+			.flags = REGL_READABLE | REGL_WRITABLE,
+		},
+		.value_register  = 0x1B,
+		.enable_register = OUTPUT_POWER_CONTROL_2,
+		.bit = 2,
+	},
+	[AXP803_REGL_FLDO1] = {
+		.info = {
+			.name      = "FLDO1",
+			.min_value = 700,
+			.max_value = 1450,
+			.ranges    = {
+				{
+					.start_raw   = 0x0B,
+					.start_value = 700,
+					.step        = 50,
+				},
+			},
+			.flags = REGL_READABLE | REGL_WRITABLE,
+		},
+		.value_register  = 0x1C,
+		.enable_register = OUTPUT_POWER_CONTROL_3,
+		.bit = 2,
+	},
+	[AXP803_REGL_FLDO2] = {
+		.info = {
+			.name      = "FLDO2",
+			.min_value = 700,
+			.max_value = 1450,
+			.ranges    = {
+				{
+					.start_raw   = 0x04,
+					.start_value = 700,
+					.step        = 50,
+				},
+			},
+			.flags = REGL_READABLE | REGL_WRITABLE,
+		},
+		.value_register  = 0x1D,
+		.enable_register = OUTPUT_POWER_CONTROL_3,
+		.bit = OUTPUT_POWER_CONTROL_3,
+	},
+	[AXP803_REGL_GPIO0LDO] = {
+		.info = {
+			.name      = "GPIO0LDO",
+			.min_value = 700,
+			.max_value = 3300,
+			.ranges    = {
+				{
+					.start_raw   = 0x1A,
+					.start_value = 700,
+					.step        = 100,
+				},
+			},
+			.flags = REGL_READABLE | REGL_WRITABLE,
+		},
+		.value_register  = 0x91,
+		.enable_register = 0x00,
+		.bit = 0x00,
+	},
+	[AXP803_REGL_GPIO1LDO] = {
+		.info = {
+			.name      = "GPIO1LDO",
+			.min_value = 700,
+			.max_value = 3300,
+			.ranges    = {
+				{
+					.start_raw   = 0x1A,
+					.start_value = 700,
+					.step        = 100,
+				},
+			},
+			.flags = REGL_READABLE | REGL_WRITABLE,
+		},
+		.value_register  = 0x93,
+		.enable_register = 0x00,
+		.bit = 0x00,
+	},
+};
+
+static uint8_t
+axp803_get_count(struct device *dev __unused)
+{
+	return AXP803_REGL_COUNT;
+}
+
+static struct regulator_info *
+axp803_get_info(struct device *dev __unused, uint8_t id)
+{
+	assert(id < REGULATOR_COUNT);
+
+	return &axp803_regl_info[id].info;
+}
+
+static int
+axp803_get_state(struct device *dev, uint8_t id)
+{
+	int     err;
+	uint8_t reg;
+
+	if ((err =
+		     i2c_read_reg(dev->bus, dev->addr,
+		                  axp803_regl_info[id].enable_register,
+		                  &reg)))
+		return err;
+
+	return (reg & BIT(0)) == 0;
+}
+
+static int
+axp803_read_raw(struct device *dev, uint8_t id, uint32_t *raw)
+{
+	int     err;
+	uint8_t reg;
+
+	if ((err =
+		     i2c_read_reg(dev->bus, dev->addr,
+		                  axp803_regl_info[id].value_register,
+		                  &reg)))
+		return err;
+	*raw = reg & ~BIT(7);
+
+	return SUCCESS;
+}
+
+static int
+axp803_set_state(struct device *dev, uint8_t id, bool enabled)
+{
+	int     err;
+	uint8_t reg;
+
+	if ((err =
+		     i2c_read_reg(dev->bus, dev->addr,
```

Comment:
These longer lines are hard to read. They can be shortened to fix the poor formatting.